index.js

```diff
@@ -233,12 +233,33 @@ var handleCloudWatch = function(event, context) {
   var region = event.Records[0].EventSubscriptionArn.split(":")[3];
   var subject = "AWS CloudWatch Notification";
   var alarmName = message.AlarmName;
-  var metricName = message.Trigger.MetricName;
+  var trigger = message.Trigger;
+  var alarmExpression;
+  if (typeof trigger.MetricName === "undefined") {
+    // This is a CloudWatch metric math alarm. Instead of a MetricName and
+    // statistic there is an array of Metrics where the first element is the
+    // math expression.
+
+    // first build a dictionary mapping metric ids to metric name and statistic
+    var sourceMetrics = {};
+    for (const metric of trigger.Metrics.slice(1)) {
+      sourceMetrics[metric.Id] = metric.MetricStat.Stat + ":"
+        + metric.MetricStat.Metric.MetricName;
+    }
+
+    // now replace each instance of the metric id in the alarm expression
+    alarmExpression = trigger.Metrics[0].Expression;
+    for (var metricid in sourceMetrics) {
```

Comment:
`for...in` loops should be avoided. If you are just trying to get the `metric.Id`s, this loop can probably be a copy of or merged with the previous.

```javascript
alarmExpression = trigger.Metrics[0].Expression;
var triggerMetricsLength = trigger.Metrics.length;
for (var i = 1; i < triggerMetricsLength; i++) {
  var metric = trigger.Metrics[i];
  alarmExpression = alarmExpression.replace(
    new RegExp(metric.Id, 'g'),
    metric.MetricStat.Stat + ':' + metric.MetricStat.Metric.MetricName
  );
}
```

Reply:
great suggestion - adopted this logic